frontend/src/components/MetacodeSelect.js

```diff
@@ -0,0 +1,168 @@
+/* global $ */
+
+import React, { Component, PropTypes } from 'react'
+
+const ENTER_KEY = 13
+const LEFT_ARROW = 37
+const UP_ARROW = 38
+const RIGHT_ARROW = 39
+const DOWN_ARROW = 40
+
+const Metacode = (props) => {
+  const { m, onClick, underCursor } = props
+  
+  return (
+    <li onClick={() => onClick(m.id) } className={ underCursor ? 'keySelect' : '' }>
+      <img src={ m.get('icon') } />
+      <span>{ m.get('name') }</span>
+    </li>
+  )
+}
+
+class MetacodeSelect extends Component {
+  
+  constructor (props) {
+    super(props)
+    this.state = {
+      filterText: '',
+      activeTab: 0,
+      selectingSection: true,
+      underCursor: 0
+    }
+  }
+  
+  componentDidMount() {
+    const self = this
+    setTimeout(function() {
+      $(document.body).on('keyup.metacodeSelect', self.handleKeyUp.bind(self))
+    }, 10)
+  }
+  
+  componentWillUnmount() {
+    $(document.body).off('.metacodeSelect')
+  }
+  
+  changeFilterText (e) {
+    this.setState({ filterText: e.target.value, underCursor: 0 })
+  }
```

Comment:
Usually I have to write these functions like

```
changeFilterText = e => {
  ...
}
```

or else this won't be bound properly. Does this function work with the regular class syntax?